core/src/commonMain/kotlin/com/saveourtool/processbuilder/ProcessBuilder.kt

```diff
@@ -44,104 +47,88 @@ class ProcessBuilder(private val useInternalRedirections: Boolean, private val f
         }
-        if (command.contains(">") && useInternalRedirections) {
+        if (config.defaultRedirects !is ProcessBuilderConfig.Redirects.None && command.contains(">")) {
             logger.error {
                 "Found user provided redirections in `$command`. " +
-                        "SAVE will create own redirections for internal purpose, please refuse redirects or use corresponding argument [redirectTo]"
+                        "SAVE will create own redirections for internal purpose, " +
+                        "please refuse redirects or use corresponding argument [redirectTo]"
             }
         }
 
         // Temporary directory for stderr and stdout (posix `system()` can't separate streams, so we do it ourselves)
-        val tmpDir = (FileSystem.SYSTEM_TEMPORARY_DIRECTORY /
-                ("ProcessBuilder_" + Clock.System.now().toEpochMilliseconds()).toPath())
-        logger.trace { "Creating temp directory: $tmpDir" }
         // Path to stdout file
-        val stdoutFile = tmpDir / "stdout.txt"
+        val stdoutFile = redirectTo.redirectStdoutTo
         logger.trace { "Creating stdout file of ProcessBuilder: $stdoutFile" }
         // Path to stderr file
-        val stderrFile = tmpDir / "stderr.txt"
+        val stderrFile = redirectTo.redirectStderrTo
         logger.trace { "Creating stderr file of ProcessBuilder: $stderrFile" }
         // Instance, containing platform-dependent realization of command execution
-        val processBuilderInternal = ProcessBuilderInternal(stdoutFile, stderrFile, useInternalRedirections)
-        fs.createDirectories(tmpDir)
-        fs.createFile(stdoutFile)
-        fs.createFile(stderrFile)
-        logger.trace { "Created temp directory $tmpDir for stderr and stdout of ProcessBuilder" }
+        val processBuilderInternal = ProcessBuilderInternal(redirectTo, config.childProcessUsername)
+
+        stdoutFile?.let { path ->
+            fs.createFile(path)
+            logger.debug { "Created file $path" }
+        }
+        stderrFile?.let { path ->
+            fs.createFile(path)
+            logger.debug { "Created file $path" }
+        }
 
         val cmd = modifyCmd(command, directory, processBuilderInternal)
 
-        logger.debug { "Executing: $cmd with timeout $timeOutMillis ms" }
-        val status = try {
-            processBuilderInternal.exec(cmd, timeOutMillis)
+        logger.debug { "Executing: $cmd with timeout $timeOutDuration ms" }
+        val executionResult = try {
+            processBuilderInternal.execute(cmd, timeOutDuration)
         } catch (ex: ProcessTimeoutException) {
-            fs.deleteRecursively(tmpDir)
+            stdoutFile?.let { fs.delete(it) }
+            stderrFile?.let { fs.delete(it) }
             throw ex
         } catch (ex: Exception) {
-            fs.deleteRecursively(tmpDir)
+            stdoutFile?.let { fs.delete(it) }
```

Comment:
What if an attempt to delete the 1st file results in an I/O exception?

Please, as a bare minimum, wrap with `try ... finally`

Reply:
The point is that if the user wants to use his/her own redirections, we might want to keep the output files if there were no exceptions. In this case `finally` is not wanted

Reply:
Ok, will try